plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -190,6 +193,10 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(ConnectorPartitionHan
                 continue;
             }
 
+            if (pathDomain.isPresent() && !pathMatchesPredicate(pathDomain.get(), hadoopPath(scanTask.file().path().toString()))) {
```

Comment:
I don't think `hadoopPath` is appropriate here. It appends some `#...` in case of `s3://`, 
but that suffix  is not -- i hope so -- returned when doing `SELECT "$path" FROM t`

Reply:
can we have some test coverage with s3?

sth like

- create a table with two files
- get `"$path"` for one of the files
- select where $path = selected_path
- verify we get data from that one file only

Reply:
I think we should use `hadoopPath()` unless fixing `$path` result. 
```java
IcebergSplitSource
    private IcebergSplit toIcebergSplit(FileScanTask task)
    {
        return new IcebergSplit(
                hadoopPath(task.file().path().toString()),
↓

IcebergPageSourceProvider
        ReaderPageSource dataPageSource = createDataPageSource(
                session,
                hdfsContext,
                new Path(split.getPath()),

...
                else if (column.isPathColumn()) {
                    columnAdaptations.add(ColumnAdaptation.constantColumn(nativeValueToBlock(FILE_PATH.getType(), utf8Slice(path.toString()))));
                }
```

It appends `#` when the bucket starts with `/` likes `s3:///bucket` (not always) as you already know (https://github.com/trinodb/trino/pull/11998)

I tried if we can create such bucket in S3 & Minio, but failing by invalid bucket name. 
```
~ aws s3api create-bucket --bucket /ebyhr-test

Parameter validation failed:
Invalid bucket name "/ebyhr-test": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN matching the regex "^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
``` 
https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html

Reply:
the `#...` suffix is internal thing, not to be exposed to users.